MeshLib/MeshSearch/NodeSearch.h

```diff
@@ -29,8 +29,9 @@ class NodeSearch final
 	/// return marked node IDs
 	const std::vector<std::size_t>& getSearchedNodeIDs() const {return _marked_nodes; }
 
-	/// Marks all nodes connecting to any of the given elements
-	std::size_t searchByElementIDs(const std::vector<std::size_t> &element_ids, bool only_match_all_connected_elements = false);
+	/// Marks all nodes connected to any of the given elements ids.
+    /// \return number of connected nodes.
+	std::size_t searchNodesConnectedToElements(const std::vector<std::size_t> &element_ids);
```

Comment:
the function does not search nodes connected to any of the given elements. it searches nodes connected  "only" to elements in the given list. therefore I suggest renaming the function to something like `searchNodesConnectedToOnlyGivenElements()` or etc.